pennylane/measure.py

```diff
@@ -255,7 +261,7 @@ def circuit(x):
             f"{op.name} is not an observable: cannot be used with expval"
         )
 
-    return MeasurementProcess(Expectation, obs=op)
+    return MeasurementProcess(Expectation, obs=op, shape=1)
```

Comment:
There seems to be a diff between the shape of the result when:
1. Passing a `QuantumTape` object to `qml.execute`;
2, Evaluating a QNode.

E.g., for `qml.state` the result in each case would be:
1. `(1,4)`;
2. `(4,)`

So QNode evaluation may include an additional squeezing step.

This functionality is meant to work as 1. This also corresponds to what JAX expects because we're executing a tape under the hood (shape comparison at the end of the line):
```
E           TypeError: Callback func <function _call.<locals>.<lambda> at 0x7f9154686430> should have returned a result with abstract values [ShapedArray(complex64[4])] but returned [ShapedArray(complex64[1,4])]
```

Example showcasing the difference:

```py
    import pennylane as qml
    from pennylane.gradients import param_shift
    from pennylane import numpy as np

    dev = qml.device("default.qubit", wires=2)
    a = np.array([0.1, 0.2, 0.3])

    def func(a):
        qml.RY(a[0], wires=0)
        qml.RX(a[1], wires=0)
        qml.RY(a[2], wires=0)

    with qml.tape.QuantumTape() as tape:
        func(a)
        qml.state()

    # 1. Execute the tape
    tape_execute_res = qml.execute(
        [tape],
        dev,
        cache=None,
        interface="autograd",
        gradient_fn=param_shift
    )

    # Get the result of the first tape execution
    tape_execute_res = tape_execute_res[0]

    # 2. Evaluate the QNode
    @qml.qnode(dev,interface='autograd')
    def circuit(a):
        func(a)
        return qml.state()

    res_qnode = circuit(a)
```
```py 
    >>> print(tape_execute_res.shape, res_qnode.shape)
    (1, 4) (4,)
```

Reply:
> There seems to be a diff between the shape of the result when:

Yeah :( This is a historical accident, and is due to the QNode attempting to squeeze the output and turn it into an ndarray. This is something I have been wanting to revisit for a long time in the 'QNode returns tuples rather than arrays' story